File: BankingAppObjectiveC/BankingApp/MicrosoftLicenseViewController.m
//
//  MicrosoftLicenseViewController.m
//  BankingApp
//
//  Created by Aasveen Kaur on 2/14/17.
//  Copyright © 2017 Jason Kim. All rights reserved.
//

#import "MicrosoftLicenseViewController.h"



@interface MicrosoftLicenseViewController ()

@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *loadingIndicator;
@property (weak, nonatomic) IBOutlet UIWebView *microsoftLicensePDFWebView;

@end

@implementation MicrosoftLicenseViewController
    NSURLRequest* request;
- (IBAction)closeButtonPressed:(UIButton *)sender {
    [self dismissViewControllerAnimated:YES completion:nil];
}
- (IBAction)licenseAccepted:(id)sender {
    NSString *key = @"AcceptedVideoLicense";
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults setBool:YES forKey:key];
    [self.delegate controller:self didAcceptLicense:YES];
    [self dismissViewControllerAnimated:YES completion:nil];
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
            _microsoftLicensePDFWebView.delegate = self;
    if( [[NSBundle mainBundle] URLForResource:@"Skype for Business App SDK Codec End User License Terms" withExtension:@"pdf"] != nil){
        request = [[NSURLRequest alloc]initWithURL:[[NSBundle mainBundle] URLForResource:@"Skype for Business App SDK Codec End User License Terms" withExtension:@"pdf"]];
    }
}

-(void)viewDidAppear:(BOOL)animated{
    [_microsoftLicensePDFWebView loadRequest:request];
    
    
}

-(void)webViewDidFinishLoad:(UIWebView *)webView{
    if (_microsoftLicensePDFWebView.isLoading)
        return
    
    [_loadingIndicator stopAnimating];

}
@end

File: BankingAppSwift/BankingAppSwift/MicrosoftLicenseViewController.swift
//
//  MicrosoftLicenseViewController.swift
//  BankingAppSwift
//
//  Created by Aasveen Kaur on 2/13/17.
//  Copyright © 2017 Aasveen Kaur. All rights reserved.
//

import UIKit

protocol MicrosoftLicenseViewControllerDelegate {
    func controller(_ controller: MicrosoftLicenseViewController , didAcceptLicense acceptedLicense:Bool)
}

class MicrosoftLicenseViewController: UIViewController,UIWebViewDelegate  {
    var delegate: MicrosoftLicenseViewControllerDelegate?
    var request:URLRequest?
    @IBOutlet weak var loadingIndicator: UIActivityIndicatorView!
    @IBAction func closeButtonPressed(_ sender: AnyObject) {
        
        dismiss(animated: true, completion: nil)
    }
    @IBOutlet weak var microsoftLicensePDFWebView: UIWebView!
    
    @IBAction func licenseAccepted(_ sender: AnyObject) {
        let key = "AcceptedVideoLicense"
        let defaults = UserDefaults.standard
        defaults.set(true, forKey: key)
        delegate?.controller(self, didAcceptLicense: true)
        dismiss(animated: true, completion: nil)
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        microsoftLicensePDFWebView.delegate = self
        if let licensePDFPath = Bundle.main.url(forResource: "Skype for Business App SDK Codec End User License Terms", withExtension: "pdf"){
             request = URLRequest(url: licensePDFPath)
            
            
        }
        
        
        
        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        microsoftLicensePDFWebView.loadRequest(request!)
        
    }
    
    func webViewDidFinishLoad(_ webView: UIWebView) {
        if (microsoftLicensePDFWebView.isLoading){
            return
        }
        loadingIndicator.stopAnimating()
        
    }
    
    
    
}
